blocks/bricks/recurrent.py

```diff
@@ -198,12 +200,22 @@ def scan_function(*args):
             outputs_info = (list(states_given.values()) +
                             [None] * (len(application.outputs) -
                                       len(application.states)))
+            if config.scan_strict:
+                # Find all shared variables
+                shared_vars = list(Selector(brick).get_params().values())
```

Comment:
We can always add `shared_vars` to `non_sequences`, regardless whether strict is `True` or false.

That I suggest that Blocks always operates in strict compatible mode, the user just has to declare explicitly that he wants to take a risk and pass `strict` to Theano.

By the way, we could make `strict` a keyword argument of `recurrent_apply`, similarly to `iterate` and `return_initial_states`. That would give nice fine-grained control to the user.